pandas/tests/indexes/test_indexing.py

```diff
@@ -96,3 +99,46 @@ def test_getitem_deprecated_float(idx):
 
     expected = idx[1]
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "idx, target, expected",
+    [
+        ([np.nan, "var1", np.nan], [np.nan], np.array([0, 2], dtype=np.int64)),
```

Comment:
~all of the expected should be np.intp~ wow we actually return np.int64, this is totally wrong, these *should* all be `np.intp` (but dont' change here, can you raise an issue for this).

Reply:
raised new issue below